app/models/manageiq/providers/vmware/infra_manager/ems_ref_obj_mixin.rb

```diff
@@ -0,0 +1,12 @@
+module ManageIQ::Providers::Vmware::InfraManager::EmsRefObjMixin
+  extend ActiveSupport::Concern
+
+  def ems_ref_obj
+    @ems_ref_obj ||= VimString.new(ems_ref, ems_ref_type, :ManagedObjectReference) if ems_ref.present?
+  end
+
+  def ems_ref=(val)
+    @ems_ref_obj = nil
+    super
+  end
```

Comment:
Can we gets direct specs for this mixin?  I know we essentially have it covered with the other specs, but I think direct specs would be beneficial

Basically, I see 4 specs...

- .ems_ref_obj when .ems_ref is nil
- .ems_ref_obj when .ems_ref is not nil
- Call .ems_ref_obj, then call .ems_ref = nil and show that .ems_ref_obj is now nil
- Call .ems_ref_obj, then call .ems_ref = <new_value> and show that .ems_ref_obj is now changed

Reply:
:+1: done https://github.com/ManageIQ/manageiq-providers-vmware/pull/465/files#diff-da9f3b510d1cafa283af41290b9094c2R1